File: java/previous/morse/src/test/java/MorseTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class MorseTest {

    private final Morse sut = new Morse();

    @Test
    public void translateRegularToMorse() {
        String expected = ".... --- .-.. .- / -- ..- -. -.. ---";

        String actual = sut.textToMorse("HOLA MUNDO");

        assertEquals(expected, actual);
    }

    @Test
    public void translateATextWithNumbersToMorse() {
        String expected = "... --- -- --- ... / ..- -. .- ... / ....- ..... / .--. . .-. ... --- -. .- ...";

        String actual = sut.textToMorse("SOMOS UNAS 45 PERSONAS");

        assertEquals(expected, actual);
    }

    @Test
    public void translateATextWithSymbolsToMorse() {
        String expected = "--...- -- .. / . -- .- .. .-.. / . ... / .--. . .--. . .--.-. .. -. --. .-.-.- -.-. --- -- -.-.--";

        String actual = sut.textToMorse("¡MI EMAIL ES PEPE@ING.COM!");

        assertEquals(expected, actual);
    }

    @Test
    public void translateAMorseRegularText() {
        String expected = "HOLA MUNDO";

        String actual = sut.morseToText(".... --- .-.. .- / -- ..- -. -.. ---");

        assertEquals(expected, actual);
    }

    @Test
    public void translateMorseToTextWithNumbers() {
        String expected = "SOMOS UNAS 45 PERSONAS";

        String actual = sut.morseToText("... --- -- --- ... / ..- -. .- ... / ....- ..... / .--. . .-. ... --- -. .- ...");

        assertEquals(expected, actual);
    }

    @Test
    public void translateMorseToTextWithSymbols() {
        String expected = "¡MI EMAIL ES PEPE@ING.COM!";

        String actual = sut.morseToText("--...- -- .. / . -- .- .. .-.. / . ... / .--. . .--. . .--.-. .. -. --. .-.-.- -.-. --- -- -.-.--");

        assertEquals(expected, actual);
    }

}

File: javascript/challenge/morse/morse.test.js
const { morseToText, textToMorse } = require("./morse.js");

describe('Morse code tests', () => {
  test('Should translate a regular text to morse', () => {
    expect(textToMorse('HOLA MUNDO')).toBe('.... --- .-.. .- / -- ..- -. -.. ---');
  });

  test('Should translate a text with numbers to morse', () => {
    expect(textToMorse('SOMOS UNAS 45 PERSONAS')).toBe('... --- -- --- ... / ..- -. .- ... / ....- ..... / .--. . .-. ... --- -. .- ...');
  });

  test('Should translate a text with symbols to morse', () => {
    expect(textToMorse('¡MI EMAIL ES PEPE@ING.COM!')).toBe('--...- -- .. / . -- .- .. .-.. / . ... / .--. . .--. . .--.-. .. -. --. .-.-.- -.-. --- -- -.-.--');
  });

  test('Should translate a morse to regular text', () => {
    expect(morseToText('.... --- .-.. .- / -- ..- -. -.. ---')).toBe('HOLA MUNDO');
  });

  test('Should translate morse to text with numbers', () => {
    expect(morseToText('... --- -- --- ... / ..- -. .- ... / ....- ..... / .--. . .-. ... --- -. .- ...')).toBe('SOMOS UNAS 45 PERSONAS');
  });

  test('Should translate morse to text with symbols', () => {
    expect(morseToText('--...- -- .. / . -- .- .. .-.. / . ... / .--. . .--. . .--.-. .. -. --. .-.-.- -.-. --- -- -.-.--')).toBe('¡MI EMAIL ES PEPE@ING.COM!');
  });  
});